### ISSUE TITLE
⭐ Add floatDiffInWeeks() method

### ISSUE THREAD

**Issue by ThaDaVos** [briannesbitt/Carbon#2175]
Hello,

I encountered an issue with the following code:
```php
echo Carbon::now()->floatDiffInWeeks(new Carbon("Last day of December this year"));
```

Carbon version: **2.32.0**

PHP version: **7.2**

I expected to get:

```
17.43
```

But I actually get:

```bash

   BadMethodCallException  : Method floatDiffInWeeks does not exist.

  at /{PROJECT_DIR}/vendor/nesbot/carbon/src/Carbon/Traits/Date.php:2588
    2584|                     }
    2585|                 }
    2586| 
    2587|                 if ($this->localStrictModeEnabled ?? static::isStrictModeEnabled()) {
  > 2588|                     throw new BadMethodCallException("Method $method does not exist.");
    2589|                 }
    2590| 
    2591|                 return null;
    2592|             }

  Exception trace:

  1   Carbon\Carbon::Carbon\Traits\{closure}()
      /{PROJECT_DIR}/vendor/nesbot/carbon/src/Carbon/Traits/Mixin.php:153

  2   Carbon\Carbon::bindMacroContext(Object(Carbon\Carbon), Object(Closure))
      /{PROJECT_DIR}/vendor/nesbot/carbon/src/Carbon/Traits/Date.php:2595

  Please use the argument -v to see more details.
```

Thanks!

**Comment by kylekatarnls** [comment]
Hello,

All the `floatDiffIn*` methods are listed in examples in the documentation. The absence of a method is not a bug, this is a feature request.

As this sounds not like a big deal to add (considering it would return the diff in days / days per week), I would agree to implement this feature.

This is open to pull-requests.

Thanks.

**Comment by kylekatarnls** [comment]
As a work-around you also can use a macro for this:
```php
Carbon::macro('floatDiffInWeeks', function (...$params) {
  return self::this()->floatDiffInDays(...$params) / Carbon::DAYS_PER_WEEK;
});

echo Carbon::now()->floatDiffInWeeks(new Carbon("Last day of December this year"));
```

**Comment by kylekatarnls** [comment]
It will be available in 2.40.0 release.
